core/field_variable.js

```diff
@@ -192,15 +187,11 @@ Blockly.FieldVariable.prototype.onItemSelected = function(menu, menuItem) {
             }
           });
       return;
-    } else if (this.deleteVarItemIndex_ >= 0 &&
-        menu.getChildAt(this.deleteVarItemIndex_) === menuItem) {
+    } else if (id == Blockly.DELETE_VARIABLE_ID) {
       // Delete variable.
       workspace.deleteVariable(this.getText());
       return;
     }
-
-    // Call any validation function, and allow it to override.
-    itemText = this.callValidator(itemText);
```

Comment:
Did you decide that validators don't make sense in the context of a variable dropdown?